components/lsp/LSP.tsx

```diff
@@ -42,33 +42,60 @@ const LSP: FC<Props> = ({
 
   const [lspContract, setLSPContract] = useState<ethers.Contract | null>(null);
   const [contractState, setContractState] = useState<ContractState>(
-    data.contractState
+    ContractState.Open
   );
 
   const [currentTime, setCurrentTime] = useState<string>("");
   const [showSettle, setShowSettle] = useState(false);
   const [settleButtonDisabled, setSettleButtonDisabled] = useState(false);
 
+  // Check if expired price has been requested
+  useEffect(() => {
```

Comment:
Do you think it would be simpler to make this useEffect and the one below a single useEffect since they're both setting the same state?

I don't think there are any race conditions in the current setup, but it might be a little difficult to reason about the interaction between the two.

Thoughts?

Reply:
yea can do